### PR TITLE
relation between forks and repos defined

### PR DESCRIPTION
**Pull request by sonalkr132**
As suggested, I have changed parent to parent_id and defined relation between forks and it's parents. `project.forks` and `project.parents` commands work from rails console. They would be handy, when we implement view for seeing all the forks of a certain project.
I had to make a few changes in view as per new project<-fork relationship.

I tend to follow thumb rule that whenever you add a migrate you try to migrate it all the way down to version 0. I came across many irreversible migrations, I have fixed some of them but eventually I had to give up and comment things out whenever reverting failed.. I would like to discuss how we should handle this irreversible migration situation here #230

### CODE REVIEW

File: app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -305,7 +304,7 @@ def fork
     child = Project.new name: @project.name,
                         user_id: current_user.id,
                         uniqueurl: @project.uniqueurl,
-                        parent: @project.parent
+                        parent_id: @project.id
```
Comment (EmilyDirsh) [inline app/controllers/projects_controller.rb]: The problem with this, and I mentioned it in my comment to #229, is that if you're forking a project that is _itself_ a fork, the parent of the new project should be the original project, not the fork. Setting a non-forked project as its own parent, and using the parent id as the parent of forks accomplished this.

File: app/models/comment.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@ class Comment < ActiveRecord::Base
   belongs_to :user
   validates :body, :presence => :true
 
-  default_scope order: 'comments.created_at DESC'
+  default_scope { order('comments.created_at DESC')}
```
Comment (EmilyDirsh) [inline app/models/comment.rb]: Why did you change the coding style here? It doesn't match the coding style in the rest of the project.

Comment (sonalkr132) [inline app/models/comment.rb]: Support for default_scope without block was removed from rails 4.1. You should look [here](http://guides.rubyonrails.org/upgrading_ruby_on_rails.html#changes-on-default-scopes#changes-on-default-scopes) and [here](http://stackoverflow.com/questions/24163571/default-scope-activerecord-4-1-block-usage-for-ordering#answer-24164710) 
Although, doc of default_scope still says we can use ones without block. But error in log clearly said that default_scope without blocks is no longer supported. I think that stackoverflow question explains it a bit.

**Comment by EmilyDirsh** [comment]
@sonalkr132 can you fix the merge conflicts and update the PR? Thanks

**Comment by sonalkr132** [comment]
Why did we need to remove:
```Ruby
issue = Issue.find(@comment.polycomment_id)
project = issue.project
```

**Comment by Alwahsh** [comment]
@sonalkr132 
I simply removed 
```
issue = Issue.find(@comment.polycomment_id)
project = issue.project
```
because I inserted another hiddent field that held the project... No need to search for it this way anymore...

**Comment by sonalkr132** [comment]
It doesn't solve the problem of ajex call not succeeding. view uses this line
```Ruby
%footer
    %p
      = link_to "Issue?", "#{@project.urlbase}/issues/new?desc=#{comment.body}"
```
Now unless you define that project variable in create method, call won't succeed.
